dashboard/test/ui/features/platform/signing_in.feature

```diff
@@ -70,3 +70,11 @@ Scenario: Join existing section from sign in page goes to section join page
   And I click ".section-sign-in button" to load a new page
   Then I wait to see ".main"
   And element ".main" contains text "Register to join section"
+
+@no_mobile
+# Ensures that the session is properly terminated during sign-out and not restored by concurrent request
+# See: https://docs.google.com/document/d/1_H7SDk1vPnUr5N7I14FPSoYs2bPPBGUz7KUAIE8X_A0
+Scenario: Teacher successfully signs in and out 10 consecutive times without any issues
```

Comment:
I'm still not able to get this test to pass locally with `bundle exec ./runner.rb --verbose --html --local -f features/platform/signing_in.feature` when running dashboard with puma, only when running the single-threaded server, and even then it's pretty flaky.

I'm worried it might also fail in other environments (like the DTT) despite passing in drone. Does it pass consistently for you when running locally? Is there some issue with my local environment that might be causing this? 

```
[platform/signing_in]   # Ensures that the session is properly terminated during sign-out and not restored by concurrent request
[platform/signing_in]   # See: https://docs.google.com/document/d/1_H7SDk1vPnUr5N7I14FPSoYs2bPPBGUz7KUAIE8X_A0
[platform/signing_in]   @no_mobile
[platform/signing_in]   Scenario: Teacher successfully signs in and out 10 consecutive times without any issues # features/platform/signing_in.feature:77
[platform/signing_in]     Given I create a teacher named "Teacher"                                              # features/step_definitions/account_steps.rb:216
[platform/signing_in]     And I sign out                                                                        # features/step_definitions/account_steps.rb:276
[platform/signing_in]     Given sign in "Teacher" 10 times                                                      # features/step_definitions/account_steps.rb:336
[platform/signing_in]
[platform/signing_in]       expected: false
[platform/signing_in]            got: true
[platform/signing_in]
[platform/signing_in]       (compared using ==)
[platform/signing_in]        (RSpec::Expectations::ExpectationNotMetError)
[platform/signing_in]       ./features/step_definitions/steps.rb:876:in `/^element "([^"]*)" is (not )?visible$/'
[platform/signing_in]       ./features/step_definitions/account_steps.rb:348:in `/^sign in "([^"]*)" (\d+) times$/'
[platform/signing_in]       ./features/support/hooks.rb:28:in `block in <top (required)>'
[platform/signing_in]       features/platform/signing_in.feature:80:in `Given sign in "Teacher" 10 times'
[platform/signing_in]
[platform/signing_in] Failing Scenarios:
[platform/signing_in] cucumber features/platform/signing_in.feature:77 # Scenario: Teacher successfully signs in and out 10 consecutive times without any issues
[platform/signing_in]
[platform/signing_in] 7 scenarios (1 failed, 6 passed)
[platform/signing_in] 59 steps (1 failed, 58 passed)
[platform/signing_in] 0m43.443s
```

Reply:
The test isn’t flaky, it's simply very effective at reproducing the issue because the solution only partially fixes it by reducing its frequency.

I’ll create a more realistic scenario where users don’t immediately sign out after signing in.